lntest/harness.go

```diff
@@ -853,14 +914,24 @@ func WaitPredicate(pred func() bool, timeout time.Duration) error {
 // a way to assert that some property is maintained over a particular time
 // frame.
 func WaitInvariant(statement func() bool, timeout time.Duration) error {
+	const pollInterval = 20 * time.Millisecond
+
 	exitTimer := time.After(timeout)
 	for {
 		select {
 		case <-exitTimer:
-			return nil
-		default:
+			// Check that the invariant holds after our requested
+			// interval expires.
+			if statement() {
+				return nil
+			}
+
+			return fmt.Errorf("invariant broken before time out")
+
+		case <-time.After(pollInterval):
```

Comment:
this could be simplified to

```go
for {
     <-time.After(pollInterval):
      if !statement() {
                return fmt.Err...
      }
      select {
       case <-exitTimer:
                  return nil
       }
}
```